src/qmflows/packages/packages.py

```diff
@@ -752,14 +762,12 @@ def ignore_unused_kwargs(fun: Callable, args: Iterable, kwargs: Mapping) -> Any:
     ps = inspect.signature(fun).parameters
 
     # Look for the arguments with the nonempty defaults.
-    defaults = list(filter(lambda t: t[1].default != inspect._empty,
-                           ps.items()))
-    # there are not keyword arguments in the function
-    if not kwargs or not defaults:
-        return fun(*args)
-    else:  # extract from kwargs only the used keyword arguments
-        d = {k: kwargs[k] for k, _ in defaults}
-        return fun(*args, **d)
+    defaults = filter(lambda t: t[1].default != inspect._empty, ps.items())
+
+    # *kwargs* may contain keyword arguments not supported by *fun*
+    # Extract from *kwargs* only the used keyword arguments
+    kwargs2 = {k: kwargs[k] for k, _ in defaults if k in kwargs}
```

Comment:
Right, I did have this problem but I forgot to fix this snippet, great that you fixed it :)